### PR TITLE
catch error thrown by nested object's getCanonicalName

### PR DESCRIPTION
**Pull request by whirakawa**
Nested object's getCanonicalName throws InternalError.

### CODE REVIEW

**Comment by xuwei-k** [comment]
Thanks. nice catch!

Should we fix some other `getCanonicalName`?  🤔 

```
scalikejdbc-core/src/main/scala/scalikejdbc/StatementExecutor.scala:                param.getClass().getCanonicalName match {
scalikejdbc-core/src/main/scala/scalikejdbc/StatementExecutor.scala:        param.getClass.getCanonicalName match {
scalikejdbc-streams/src/main/scala/scalikejdbc/streams/DatabaseSubscription.scala:          log.info(s"Failed to close the occupied database session because ${e.getMessage}, exception: ${e.getClass.getCanonicalName}")
scalikejdbc-streams/src/main/scala/scalikejdbc/streams/DatabaseSubscription.scala:                    log.info(s"Unexpectedly failed to deal with remaining iterator because ${e.getMessage}, exception: ${e.getClass.getCanonicalName}")
```

**Comment by seratch** [comment]
Fixing those seems to be safer. @whirakawa I'm sorry to bother you but could you work on the four lines Kenji mentioned as well?

**Comment by whirakawa** [comment]
OK. I simply changed getCanonicalName to getName.

**Comment by xuwei-k** [comment]
https://github.com/scala/bug/issues/2034

**Comment by seratch** [comment]
Sorry, I closed this pull request by mistake.

**Comment by seratch** [comment]
@whirakawa IHMO, we should still chose getCanonicalName as the primary API to fetch valid class name.

see also: https://stackoverflow.com/questions/15202997/what-is-the-difference-between-canonical-name-simple-name-and-class-name-in-jav

I prefer having a wrapper method like this:
```scala
def getClassName(clazz: Class): String = {
  val canonicalName: Option[String] = try {
    Option(clazz.getCanonicalName)
  } catch {
    case e: InternalError if e.getMessage == "Malformed class name" => None
  }
  canonicalName match {
    case Some(className) => className
    case _ => clazz.getName
  }
}
```

**Comment by whirakawa** [comment]
I fixed it.

File: scalikejdbc-core/src/main/scala/scalikejdbc/ClassNameUtil.scala [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package scalikejdbc
+
+object ClassNameUtil {
```
Comment (seratch) [inline scalikejdbc-core/src/main/scala/scalikejdbc/ClassNameUtil.scala]: Thanks, let me change this object to `private[scalikejdbc]` after merging your PR.

**Comment by whirakawa** [comment]
Thank you for reviewing and merging 😄

**Pull request merged**